src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/StorePal.Find.cs

```diff
@@ -0,0 +1,553 @@
+            {
+                if (filter != null && !filter(pCertContext))
+                    continue;
+
+                if (validOnly)
+                {
+                    if (!VerifyCertificateIgnoringErrors(pCertContext))
+                        continue;
+                }
+
+                if (!Interop.crypt32.CertAddCertificateLinkToStore(findResults, pCertContext, CertStoreAddDisposition.CERT_STORE_ADD_ALWAYS, IntPtr.Zero))
+                    throw new CryptographicException(Marshal.GetLastWin32Error());
+            }
+
+            return new StorePal(findResults);
+        }
+
+        private static bool VerifyCertificateIgnoringErrors(SafeCertContextHandle pCertContext)
+        {
+            ChainPal chainPal = ChainPal.BuildChain(
+                true,
+                CertificatePal.FromHandle(pCertContext.DangerousGetHandle()),
+                null, //extraStore
+                null, //applicationPolicy
+                null, //certificatePolicy
+                X509RevocationMode.NoCheck,
+                X509RevocationFlag.ExcludeRoot,
+                DateTime.Now,
+                new TimeSpan(0, 0, 0));
+
+            if (chainPal == null)
+                return false;
+
+            using (chainPal)
+            {
+                Exception verificationException;
+                bool? verified = chainPal.Verify(X509VerificationFlags.NoFlag, out verificationException);
+                if (!(verified.HasValue && verified.Value))
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static String GetCertNameInfo(SafeCertContextHandle pCertContext, CertNameType dwNameType, CertNameFlags dwNameFlags)
+        {
+            Debug.Assert(dwNameType != CertNameType.CERT_NAME_ATTR_TYPE);
+
+            CertNameStringType stringType = CertNameStringType.CERT_X500_NAME_STR | CertNameStringType.CERT_NAME_STR_REVERSE_FLAG;
+
+            int cch = Interop.crypt32.CertGetNameString(pCertContext, dwNameType, dwNameFlags, ref stringType, null, 0);
+            if (cch == 0)
+                throw new CryptographicException(Marshal.GetLastWin32Error());
+
+            StringBuilder sb = new StringBuilder(cch);
+            if (0 == Interop.crypt32.CertGetNameString(pCertContext, dwNameType, dwNameFlags, ref stringType, sb, cch))
+                throw new CryptographicException(Marshal.GetLastWin32Error());
+
+            return sb.ToString();
+        }
+
+        private static BigInteger PositiveBigIntegerFromByteArray(byte[] bytes)
+        {
+            // To prevent the big integer from misinterpreted as a negative number, add a "leading 0" to the byte array.
+            // Since BigInteger(bytes[]) requires a little-endian byte array, the "leading 0" actually goes at the end of the array.
+            byte[] newBytes = new byte[bytes.Length + 1];
```

Comment:
Are there any quick checks that could be done on ```bytes``` to determine whether it would be interpreted as negative?  That would let us avoid the unnecessary array allocation and copy.